code/game/jobs/whitelist.dm

```diff
@@ -42,12 +41,7 @@ var/global/list/alien_whitelist = list()
 		return TRUE
 
 	//If we have a loaded file, search it
-	if(alien_whitelist)
-		for (var/s in alien_whitelist)
-			if(findtext(s,"[M.ckey] - [species.name]"))
-				return TRUE
-			if(findtext(s,"[M.ckey] - All"))
-				return TRUE
+	return findtext(alien_whitelist, regex("[M.ckey] - [species.name]", "g"))

```

Comment:
I don't think this will be dangerous in practical use and it's the easiest/fastest way to go about it as mentioned in code without pre-parsing - but `ckey(M.ckey)` may be preferable for paranoia since the ckey of a mob is mutable and stuff like the @ symbol is introduced to ckeys for some admin stuff like aghosting.

The second part of the regex is harder. We can't be sure someone won't put a `.` (for example) in a species (or language) name later, or some other symbol with special regex meaning that should be escaped but won't be when templated in verbatim.